### PR TITLE
[SRE] Delay the creation of the MonoClass for dynamic types until Cre…

### PR DESCRIPTION
**Pull request by vargaz**
…ateType () is called. Remove some icalls.

### CODE REVIEW

**Comment by vargaz** [comment]
@monojenkins build

**Comment by vargaz** [comment]
build

**Comment by akoeplinger** [comment]
build

File: mono/metadata/sre.c [hunk 1]
```diff
@@ -3654,52 +3666,6 @@ ves_icall_TypeBuilder_create_runtime_class (MonoReflectionTypeBuilder *tb)
...
-			gparam->tbuilder->generic_container = (MonoGenericContainer *)mono_image_alloc0 (klass->image, sizeof (MonoGenericContainer));
-			gparam->tbuilder->generic_container->owner.klass = klass;
-		}
-	}
-
-	return TRUE;
-}
-
-void
-ves_icall_GenericTypeParameterBuilder_initialize (MonoReflectionGenericParam *gparam)
```
Comment (akoeplinger) [inline mono/metadata/sre.c]: There's another instance of this function left inside the DISABLE_REFLECTION_EMIT ifdef.

**Pull request merged**